tests/test_ranges.py

```diff
@@ -182,5 +183,148 @@ def test_can_validate_with_multi_range(self):
+        try:
+            ranges.DecimalRange(description)
+            self.fail("test must fail with InterfaceError")
+        except errors.InterfaceError as anticipated_error:
+            dev_test.assert_fnmatches(self, str(anticipated_error), anticipated_error_message_pattern)
+
+    def test_broken_ranges(self):
+        self.assertRaises(errors.InterfaceError, ranges.DecimalRange, "x")
+        self.assertRaises(errors.InterfaceError, ranges.DecimalRange, "...")
+        self.assertRaises(errors.InterfaceError, ranges.DecimalRange, "-")
+        self.assertRaises(errors.InterfaceError, ranges.DecimalRange, "-...")
+        self.assertRaises(errors.InterfaceError, ranges.DecimalRange, "1 x")
+        self.assertRaises(errors.InterfaceError, ranges.DecimalRange, "-x")
+        self.assertRaises(errors.InterfaceError, ranges.DecimalRange, "1 2")
+        self.assertRaises(errors.InterfaceError, ranges.DecimalRange, "1...2 3")
+        self.assertRaises(errors.InterfaceError, ranges.DecimalRange, "1...2-3")
+        self.assertRaises(errors.InterfaceError, ranges.DecimalRange, "1...2...3")
+        self.assertRaises(errors.InterfaceError, ranges.DecimalRange, "2...1")
+        self.assertRaises(errors.InterfaceError, ranges.DecimalRange, "2...-3")
+        self.assertRaises(errors.InterfaceError, ranges.DecimalRange, "-1...-3")
+        self._test_fails_with_interface_error(
+            '?', "range must be specified using integer numbers and ellipsis (...) but found: '?'*")
+
+    def test_fails_on_parse_hex_range(self):
+        self.assertRaises(errors.InterfaceError, ranges.DecimalRange, "0x7f")
+        self.assertRaises(errors.InterfaceError, ranges.DecimalRange, "0x7F")
+
+    def test_can_parse_multiple_ranges(self):
+        self.assertEquals(ranges.DecimalRange("1.1, 3").items, [(decimal.Decimal('1.1'), decimal.Decimal('1.1')), (3, 3)])
+        self.assertEquals(ranges.DecimalRange("1...2.3, 5...").items, [(1, decimal.Decimal('2.3')), (5, None)])
+
+    def test_fails_on_parse_symbolic_range(self):
+        self.assertRaises(errors.InterfaceError, ranges.DecimalRange, "TAB")
+        self.assertRaises(errors.InterfaceError, ranges.DecimalRange, "vt")
+        self.assertRaises(errors.InterfaceError, ranges.DecimalRange, "Tab...Vt")
+        self.assertRaises(errors.InterfaceError, ranges.DecimalRange, "Tab...11")
+
+    def test_fails_on_parse_text_range(self):
+        self.assertRaises(errors.InterfaceError, ranges.DecimalRange, "\"a\"")
+
+    def test_can_use_default_range(self):
+        self.assertEquals(ranges.DecimalRange("", "2.1...3").items, [(decimal.Decimal('2.1'), 3)])
+
+    def test_can_override_default_range(self):
+        self.assertEquals(ranges.DecimalRange("1.1...2", "2...3").items, [(decimal.Decimal('1.1'), 2)])
+
+    def test_can_get_lower_limit(self):
+        self.assertEquals(ranges.DecimalRange("5...9").lower_limit, 5)
```

Comment:
Use actual decimal ranges here, not only integer ranges. The same applies to `test_can_get_upper_limit` and `test_fails_on_inconsistent_overlapping_multi_range`.